src/pfe/file-watcher/server/src/controllers/projectsController.ts

```diff
@@ -585,32 +579,24 @@ async function triggerBuild(project: BuildQueueType, changedFiles?: IFileChangeE
  */
 function checkInProgressBuilds(): void {
     lock.acquire("runningBuildsLock", async done => {
-        // we must assert here that the running builds queue is always unique
-        // we do this by converting the running build queue into a set and making sure the length matches
-        if (buildQueue.length > 0) {
-            logger.assert(new Set(runningBuilds).size === runningBuilds.length, "Builds in running queue must be unique");
-        }
-
         // filter out all projects that are completed and reduce the build in progress by 1 for each such project
-        runningBuilds = runningBuilds.filter((project: BuildQueueType) => {
+        runningBuilds.forEach((project: BuildQueueType) => {
             const projectID = project.operation.projectInfo.projectID;
             const buildStatus = statusController.getBuildState(projectID);
             if (buildStatus === statusController.BuildState.success || buildStatus === statusController.BuildState.failed) {
                 logger.logProjectInfo("Build completed for " + projectID, projectID);
-                return false;
-            } else {
-                return true;
+                runningBuilds.delete(project);
             }
         });
 
-        if (runningBuilds.length > 0) {
+        if (runningBuilds.size > 0) {
             // Only log the runningBuilds details if there is a project building to avoid spamming the logs
             const currentBuilds: Array<String> = [];
-            for (let i = 0; i < runningBuilds.length; i++) {
-                currentBuilds.push(runningBuilds[i].operation.projectInfo.projectID);
-            }
+            runningBuilds.forEach((project: BuildQueueType) => {
```

Comment:
Do we need `currentBuilds`? Looks like it's only used for the log message below. Can we use `runningBuilds` directly for the message instead?

Reply:
The reason we use `currentBuilds` is because when we log the running builds queue, we only show the `projectID` instead of the entire build object which is of type BuildQueueType. `runningBuilds` is also of type BuildQueueType so we just take the projectID from each project and print them

Reply:
The extra information could be beneficial and it's for debug logs so won't pollute the logs by default. But I'm okay if you'd rather have only the projectID.